packages/local-cli/util/loadMetroConfig.js

```diff
@@ -135,6 +132,29 @@ module.exports = async function load(
     defaultConfig
   );
 
+  /**
+   * Metro configuration requires the following two properties to be defined
+   * statically. They can't be a function, unline `serializer.getModulesRunBeforeMainModule`.
+   *
+   * We only resolve React Native files if no other value was provided by the user.
+   *
+   * This is to allow using CLI inside React Native source code, where there is no
+   * "react-native" module in "node_modules".
+   *
+   * See "react-native/metro.config.js"
+   */
+  if (!config.resolver.hasteImplModulePath) {
```

Comment:
The idea is when React Native provides its "defaults" in "metro.config.js", this code will not execute and hence, not cause an error.

I was a bit confused coming up with a good documentation for that code (lines 135-145) - if anyone feels like this can be described in a better way, please comment.

Reply:
Couldn't find the `react-native/metro.config.js` file you're talking about, can you link to it?